app/models/auth_token.rb

```diff
@@ -2,32 +2,33 @@ class AuthToken < ActiveRecord::Base
 
   belongs_to :user
 
-  validates :authentication_token, presence: true, uniqueness: true
+  validates :encrypted_authentication_token, presence: true, uniqueness: true
 
   # Auth token encryption settings
-  attr_encrypted :auth_token,
+  attr_encrypted :authentication_token,
     key: Doubtfire::Application.secrets.secret_key_attr,
-    encode: true,
-    attribute: 'authentication_token'
+    mode: :per_attribute_iv,
+    algorithm: 'aes-256-gcm'
+
 
   def self.generate(user, remember, expiry_time = Time.zone.now + 2.hours)
     # Loop until new unique auth token is found
     token = loop do
       token = Devise.friendly_token
-      break token unless AuthToken.find_by_auth_token(token)
+      break token unless AuthToken.find_by_encrypted_authentication_token(token)
```

Comment:
Get the token from the user here as well - we want to make sure the token is unique for the user.